### DISCUSSION TITLE
RoundRobinPartition mode：enable batching and has key , What will be the run result?

### DISCUSSION THREAD

**Discussion by startjava** [apache/pulsar#18366] [Q&A] [answered]
https://pulsar.apache.org/docs/next/concepts-messaging/#routing-modes
RoundRobinPartition mode 。

**Comment by merlimat** [comment, marked as answer]
If a key is provided on the message, that it will be honored. The round-robin is only applied if there's no key.

You can find the logic in https://github.com/apache/pulsar/blob/master/pulsar-client/src/main/java/org/apache/pulsar/client/impl/RoundRobinPartitionMessageRouterImpl.java#L74
